### DISCUSSION TITLE
Meltano HackDay project ideas: April 14 2023

### DISCUSSION THREAD

**Discussion by aaronsteers** [meltano/meltano#7086] [General]
Opening this discussion to explore project ideas for our Jan 2023 HackDay event:

- https://github.com/meltano/internal-general/issues/461

**Comment by edgarrmondragon** [comment]
A few ideas:

- A tutorial for exporting and visualizing Singer metrics in a time-series database like InfluxDB
- A PoC for the Hub API built on top of FastAPI
- `singer-rust` PoC and build a simple tap or target, publish a Python wheel
- ~A node wrapper for npm plugins that uses https://pypi.org/project/nodejs-bin/~. Ken already did this for https://github.com/meltanolabs/evidence-ext

**Comment by WillDaSilva** [comment]
- Meltano shim package that comes with the real Meltano wheel, Poetry lock file (n.b. `pip freeze` cannot be used because it does not work cross-platform), and an install script.
  - This pattern could be used to install Meltano with Homebrew, APT, Yum, etc.
  - This would make it easy for Meltano to upgrade/downgrade/reset its own installation.
  - With this, Meltano would manage its own virtual environment. This eliminates the classes of issue caused by:
    - running Meltano in a dirty environment
    - installing Meltano with different package versions than anticipated
- Tap that uses OpenAI Whisper to extract transcriptions/translations from video/audio files (originally suggested by @aaronsteers [on Slack](https://meltano.slack.com/archives/CFG3C3D1Q/p1667606405206769))
- Make the `Canoncial` class lazy-load its children, to prevent cases where thousands of instances are created when only a single attribute is needed (e.g. the `version` attribute of a `MeltanoFile` instance). This ought to yield a significant performance improvement across all non-trivial Meltano commands.
- Update the `venv_service` module to provide its own Python installations, rather than relying on the Python installation that's running Meltano. This would eliminate the problem where a plugin doesn't support the Python version that's currently used. It would also enable us to e.g. use PyPy or other fast Python implementations in some cases, which may provide significant performance improvements.

**Comment by kgpayne** [comment]
- `target-singerlake` based around the spec worked on with Alex and documented [here](https://blog.kenpayne.co.uk/blog/2023-01-17.html)
- @sbalnojan and I also talked briefly about an automated samples framework for generating both the content (i.e. web-based landing page) _and_ testing the ongoing success of common tap+target pairs.
- Write up the `target-duckpond` idea for a local datalake testbed/"in-a-box"
